src/Http/Session/Session.php

```diff
@@ -0,0 +1,260 @@
+    {
+        if (array_key_exists(self::REQUEST_DATA, $session) && $session[self::REQUEST_DATA] !== null) {
+            $this->requestData->setAll($session[self::REQUEST_DATA]);
+        }
+    }
+
+    /**
+     * @param string $item
+     *
+     * @return mixed|false
+     */
+    public function get(string $item)
+    {
+        return $this->read(self::APP, $item);
+    }
+
+    /**
+     * @param string $item
+     *
+     * @param $value
+     */
+    public function add(string $item, $value): void
+    {
+        $data        = $this->read(self::APP);
+        $data[$item] = $value;
+
+        $this->write(self::APP, $data);
+    }
+
+    /**
+     * @param string $item
+     *
+     * @return bool
+     */
+    public function remove(string $item): bool
+    {
+        $data = $this->read(self::APP);
+        if (array_key_exists($item, $data)) {
+            unset($data[$item]);
+            $this->write(self::APP, $data);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $item
+     *
+     * @return bool
+     */
+    public function has(string $item): bool
+    {
+        return $this->read(self::APP, $item) != false;
+    }
+
+    /**
+     * @return array|null
+     */
+    public function getAll(): ?array
+    {
+        return $this->read(self::APP);
+    }
+
+    /**
+     * @param array $data
+     */
+    public function addAll(array $data): void
+    {
+        $this->write(self::APP, $data);
+    }
+
+    public function removeAll(): void
+    {
+        $this->write(self::APP, []);
+    }
+
+    /**
+     * @return UserInterface|null
+     */
+    public function getUser(): ?UserInterface
+    {
+        return $this->read(self::USER);
+    }
+
+    /**
+     * @param UserInterface $user
+     */
+    public function setUser(UserInterface $user): void
+    {
+        $this->write(self::USER, $user);
+    }
+
+    /**
+     * @return bool
+     */
+    public function unsetUser(): bool
+    {
+        if ($this->hasUser()) {
+            $this->write(self::USER, null);
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasUser(): bool
+    {
+        return $this->read(self::USER) instanceof UserInterface;
+    }
+
+    /**
+     * @return array
+     */
+    private function getFromGlobal(): array
+    {
+        $session = $this->read();
+        $this->write(Session::FLASHES, null);
+        $this->write(Session::REQUEST_DATA, null);
+
+        return $session;
+    }
+
+    /**
+     * @param string $container
+     * @param mixed  $data
+     */
+    private function write(string $container, $data): void
+    {
+        $_SESSION[$container] = $data;
+    }
+
+    /**
+     * @param string|null $container
+     * @param string|null $key
+     *
+     * @return mixed
+     */
+    private function read(string $container = null, string $key = null)
+    {
+        $session = $_SESSION;
```

Comment:
Codacy found an issue: [Direct use of $_SESSION Superglobal detected.](https://app.codacy.com/gh/pierregaimard/framework3/pullRequest?prid=6299285)